tests/test_effects.py

```diff
@@ -214,6 +214,31 @@ def test_stars(self):
         event = object()
         self.assertEqual(event, effect.process_event(event))
 
+    def test_stars_pattern(self):
+        """
+        Check that Stars custom pattern value works works.
+        """
+        # Check that Stars randomly updates the Screen every frame.
+        screen = MagicMock(spec=Screen, colours=8, unicode_aware=False)
+        canvas = Canvas(screen, 10, 40, 0, 0)
+        effect = Stars(canvas, 100, "TESTTESTTEST")
+        effect.reset()
+        self.assert_blank(canvas)
+        my_buffer = [[(32, 7, 0, 0) for _ in range(40)] for _ in range(10)]
+        for i in range(10):
+            effect.update(i)
+            self.assertTrue(self.check_canvas(
+                canvas,
+                my_buffer,
+                lambda value: self.assertIn(chr(value[0]), " TES")))
+
+        # Check there is no stop frame by default.
```

Comment:
There is no need to duplicate the rest of the tests in this function as they will not be affected by the custom pattern.  I'm trying to get more focused in what each test actually checks these days...  :-)